apps/omg_watcher/lib/exit_processor/core.ex

```diff
@@ -782,4 +827,99 @@ defmodule OMG.Watcher.ExitProcessor.Core do
+        utxo_exit,
+        exit_id
+      ) do
+    {%Transaction.Signed{raw_tx: challenging_tx}, input_index} =
+      get_spending_transaction_with_index(challenging_signed, utxo_exit)
+
+    %Challenge{
+      exit_id: exit_id,
+      input_index: input_index,
+      txbytes: challenging_tx |> Transaction.encode(),
+      sig: find_sig(challenging_signed, owner)
+    }
+  end
+
+  @doc """
+  Checks whether database responses hold all the relevant data successfully fetched:
+   - a block number which can be used to retrieve needed information to challenge or if exists ife which spends inputs
+   - the relevant exit information
+  """
+  @spec get_challange_data(tuple, Utxo.Position.t(), t()) ::
+          {:ok, pos_integer() | Transaction.Signed.t(), ExitInfo.t()} | {:error, atom()}
+  def get_challange_data(spending_blknum_response, exiting_utxo_pos, %__MODULE__{} = state) do
+    ife_response = get_ife_based_on_utxo(exiting_utxo_pos, state)
+    exit_response = get_exit_info(exiting_utxo_pos, state)
+    ensure_challengeable(spending_blknum_response, exit_response, ife_response)
+  end
+
+  @spec ensure_challengeable(tuple(), {:ok, ExitInfo.t() | :not_found}, {:ok, InFlightExitInfo.t() | :not_found}) ::
+          {:ok, pos_integer() | Transaction.Signed.t(), ExitInfo.t()} | {:error, atom()}
+  defp ensure_challengeable(spending_blknum_response, exit_response, ife_response)
+
+  defp ensure_challengeable({:ok, :not_found}, _, {:ok, :not_found}), do: {:error, :utxo_not_spent}
+  defp ensure_challengeable(_, {:ok, :not_found}, _), do: {:error, :exit_not_found}
+
+  defp ensure_challengeable({:ok, blknum}, {:ok, exit_info}, _) when is_integer(blknum),
+    do: {:ok, blknum, exit_info}
+
+  defp ensure_challengeable(_, {:ok, exit_info}, {:ok, %Transaction.Signed{} = signed_tx}),
+    do: {:ok, signed_tx, exit_info}
+
+  defp ensure_challengeable({:error, error}, _, _), do: {:error, error}
+  defp ensure_challengeable(_, {:error, error}, _), do: {:error, error}
+  defp ensure_challengeable(_, _, {:error, error}), do: {:error, error}
+
+  # finds transaction in given block and input index spending given utxo
+  @spec get_spending_transaction_with_index(Block.t() | Transaction.Signed.t(), Utxo.Position.t()) ::
+          {Transaction.Signed.t(), non_neg_integer()} | false
+  defp get_spending_transaction_with_index(%Block{transactions: txs}, utxo_pos) do
+    txs
+    |> Enum.map(&Transaction.Signed.decode/1)
+    |> Enum.find_value(fn {:ok, %Transaction.Signed{} = tx_signed} ->
+      # `Enum.find_value/2` allows to find tx that spends `utxo_pos` and return it along with input index in one run
+
+      get_spending_transaction_with_index(tx_signed, utxo_pos)
+    end)
+  end
+
+  defp get_spending_transaction_with_index(%Transaction.Signed{raw_tx: tx} = tx_signed, utxo_pos) do
+    inputs = Transaction.get_inputs(tx)
+
+    if input_index = Enum.find_index(inputs, &(&1 == utxo_pos)) do
+      {tx_signed, input_index}
+    else
+      false
```

Comment:
why not nil?
ps.
```
if expr, 
    do: value1,
  else: value2
```

Reply:
I went with `false` as all functions in this file went with `false` convention instead `nil` as return value

Reply:
and false is spending transaction with index?
function name suggests that null is returned or there will throw an error, in case no spending transaction find.